### PR TITLE
GH-2069: Same factory for retryable and normal endpoints

### PR DESCRIPTION
**Pull request by tomazfernandes**
Resolves https://github.com/spring-projects/spring-kafka/issues/2069

I used @garyrussell's suggestion of creating an isRetryable method in the container, and also allowing composite containerCustomizers.

I wasn't sure whether to add the composite customizer method to the docs, seemed like an unnecessary complexity if we're keeping the regular setContainerCustomizer method. But I can add it if you think it's better.

Let me know if there's anything you'd like me to change.

Thanks!

### CODE REVIEW

**Comment by tomazfernandes** [comment]
Hi, I see you're probably busy with the upcoming major release. I'd like to take advantage of the fact that this PR hasn't been yet reviewed and propose that we use #2108 instead. It changes a lot less existing code and should be easier to reason about and review.

Of course, we can reopen this one if you prefer this solution, no worries. And no hurry, if this doesn't get reviewed in time for 2.8.3 I can change the "since" tags and anything else needed.

Thanks!

File: spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java [hunk 1]
```diff
@@ -341,6 +343,25 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
 		this.retryTemplate = retryTemplate;
 	}
 
+	/**
+	 * Sets whether this container is part of a non-blocking retries pattern.
```
Comment (garyrussell) [inline spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java:347]: Strictly speaking "...whether the container created for this endpoint..."

File: spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java [hunk 2]
```diff
@@ -341,6 +343,25 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
...
+	public void setRetryable(boolean retryable) {
+		this.retryable = retryable;
+	}
+
+	/**
+	 * Whether this container is part of a non-blocking retries pattern.
+	 * @return true if it is part of a non-blocking retries pattern.
+	 * @since 2.8.3
+	 */
+	@Override
```
Comment (garyrussell) [inline spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java:360]: No need to duplicate the javadoc here.

**Review by garyrussell — COMMENTED** [review]
LGTM; thanks! Just some minor javadoc polishing.

**Comment by garyrussell** [comment]
Oops - sorry - reviewed the wrong one 😄

**Pull request closed without merging**